velox/expression/tests/ExprTest.cpp

```diff
@@ -760,6 +765,36 @@ TEST_F(ExprTest, csePartialEvaluation) {
   assertEqualVectors(expected, results[1]);
 }
 
+TEST_F(ExprTest, csePartialEvaluationWithEncodings) {
+  auto data = makeRowVector(
+      {wrapInDictionary(
+           makeIndicesInReverse(5),
+           wrapInDictionary(
+               makeIndicesInReverse(5),
+               makeFlatVector<int64_t>({0, 10, 20, 30, 40}))),
+       makeFlatVector<int64_t>({3, 33, 333, 3333, 33333})});
+
+  auto exprSet = compileMultiple(
+      {
+          "concat(concat(cast(c0 as varchar), ',', cast(c1 as varchar)), 'xxx')",
+          "concat(concat(cast(c0 as varchar), ',', cast(c1 as varchar)), 'yyy')",
```

Comment:
fyi this can also be triggered with a single expression:

"if(c1 >= 0 and c0 >= 0, if(c1 >= 0 and c0 >= 0, 1, 2), 3)"